src/surfaces/rail/rail.jsx

```diff
@@ -1,32 +1,96 @@
-
 import ClassNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
+import withStyles from '../../styles/withStyles';
+
+const propTypes = {
+    children: PropTypes.node,
+    classes: PropTypes.shape({
+        root: PropTypes.string,
+        positionLeft: PropTypes.string,
+        positionRight: PropTypes.string,
+    }),
+    className: PropTypes.string,
+    position: PropTypes.oneOf(['left', 'right']),
+};
+
+const defaultProps = {
+    children: null,
+    classes: null,
+    className: null,
+    position: 'left',
+};
 
+const styles = (theme) => ({
+    positionLeft: {},
+    positionRight: {},
+    root: {
+        backgroundColor: theme.palette.background.primary,
+        height: '100%',
+        padding: [[33, 22]],
+        width: 250,
+        '&$positionLeft, &$positionRight': {
+            '&::after': {
+                content: '""',
+                height: '100%',
+                position: 'absolute',
+                top: 0,
+                width: 15,
+            },
+        },
+        '&$positionLeft': {
+            boxShadow: `inset -1px 0 0 0 ${theme.palette.border.primary}`,
+            left: 0,
+            top: 0,
+            '&::after': {
+                background: 'linear-gradient(to right, rgba(0, 0, 0, .1) -100%, transparent 100%)',
+                right: -15,
+            },
+        },
+        '&$positionRight': {
+            boxShadow: `inset 1px 0 0 0 ${theme.palette.border.primary}`,
+            right: 0,
+            top: 0,
+            '&::after': {
+                background: 'linear-gradient(to left, rgba(0, 0, 0, .1) -100%, transparent 100%)',
+                left: -15,
+            },
+        },
+    },
+});
+
+// eslint-disable-next-line react/prefer-stateless-function
 class Rail extends React.Component {
```

Comment:
Is there a reason we're _not_ using a functional component here?

Reply:
The only reason is because it's an old component. Do you think we should take the time - probably not much, that i know - to convert?

Reply:
I think we could/should.  But if you'd rather hold off for the sake of expediency, I can understand that decision as well.

Reply:
Fixed.

Reply:
Beauty.  Resolving.